src/main/webapp/app/shared/util/core-drive-annotation-submission/core-drive-annotation-submission.ts

```diff
@@ -0,0 +1,136 @@
+import _ from 'lodash';
+import { Drug, Gene, Mutation, Review, Treatment, Tumor, Vus, DrugCollection } from '../../model/firebase/firebase.model';
+
+export function getGeneData(geneData: Gene, excludeComments: boolean, onlyReviewedContent: boolean, drugList: DrugCollection): true | Gene {
+  const gene = _.cloneDeep(geneData);
+  processData(gene, ['summary', 'background'], excludeComments, onlyReviewedContent);
+  processData(gene.type, ['tsg', 'ocg'], excludeComments, onlyReviewedContent);
+  const tempMutations: Mutation[] = [];
+  for (const mutation of gene.mutations ?? []) {
+    const tempTumors: Tumor[] = [];
+    if (shouldExclude(onlyReviewedContent, mutation.name_review)) {
+      tempMutations.push(mutation);
+      continue;
+    }
+    processData(mutation, ['name'], excludeComments, onlyReviewedContent);
+    processData(mutation.mutation_effect, ['oncogenic', 'effect', 'description'], excludeComments, onlyReviewedContent);
+    for (const tumor of mutation.tumors ?? []) {
+      if (shouldExclude(onlyReviewedContent, tumor.cancerTypes_review)) {
+        tempTumors.push(tumor);
+        continue;
+      }
+      // process tumor cancerTypes
+      processData(tumor, ['summary', 'diagnosticSummary', 'prognosticSummary'], excludeComments, onlyReviewedContent);
+      processData(tumor.diagnostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      processData(tumor.prognostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      for (const ti of tumor.TIs) {
+        processData(ti, ['description'], excludeComments, onlyReviewedContent);
+        type TempTreatment = Omit<Treatment, 'name'> & { name: ReturnType<typeof drugUuidToDrug> | string };
+        const tempTreatments: TempTreatment[] = [];
+        for (const treatment of ti.treatments ?? []) {
+          if (shouldExclude(onlyReviewedContent, treatment.name_review)) {
+            tempTreatments.push(treatment);
+            return true;
+          }
+          (treatment as TempTreatment).name = drugUuidToDrug(treatment.name, drugList);
+          processData(
+            treatment,
+            ['level', 'propagation', 'propagationLiquid', 'indication', 'description'],
```

Comment:
Is fdaLevel not here?

Reply:
It's copied from the [old code](https://github.com/oncokb/curation-platform/blob/master/app/scripts/services/mainutils.js#L481-L488)

Reply:
It only impacts if the last reviewed data is used, but my test don't break if I change it. I'm going to adjust my tests so they will break if I make changes to this logic though. I'll also add `fdaLevel` to the array.